File: 파이썬/5.에라토스테네스의 체/2581.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/2581
 '''

m = int(input())
n = int(input())
l = [True] * 10001

l[1] = False
for i in range(2, 10001) :
    if l[i] == True :
        for j in range(i*2, 10001, i) : 
            l[j] = False

res = []
for i in range(m, n+1) :
    if l[i] == True and i >= m and i <= n:
        res.append(i)

if len(res) != 0 :
    print(sum(res))
    print(min(res))
else :
    print(-1)

File: C++/7.1 에라토스테네스의 체/2581.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/2581
*/

#include <iostream>
#include <algorithm>
#include <cmath>

using namespace std;

int m, n;
bool check[10001];

int main(){
    
    cin >> m >> n;
    
    fill(check, check+10001, true);
    check[1] = false;
    
    int s = sqrt(n);
    for(int i=2; i<=s; i++){
        if(check[i]){
            for(int j=i+i; j<=n; j+=i){
                check[j] = false;
            }
        }
    }
    
    int sum = 0;
    int min_number = 10000;
    
    for(int i=m; i<=n; i++){
        if(check[i]){
            sum += i;
            min_number = min(min_number, i);
        }
        
    }
    
    if(sum != 0){
        cout << sum << "\n";
        cout << min_number;
        
    }else{
        cout << -1;
    }
    
    return 0;
}